### PR TITLE
Add a state field to RawImageFile and improved messaging for partial update of RawImageUploadSession API

### PR DESCRIPTION
**Pull request by haimasree**
Closes: #1087 
Closes: #1088

### CODE REVIEW

**Comment by jmsmkn** [comment]
I think that the `RawImageUploadSession` stores the job state, so this information does not need to be duplicated. What we really need to know is if the `RawImageFile` has either been successfully consumed into an `Image` or not, so a boolean field with a default of `False` is what I think we need.

**Comment by jmsmkn** [comment]
It might be worth combining this with #1089 into 1 PR as they're closely related.

File: app/tests/cases_tests/test_api.py [hunk 1]
```diff
@@ -295,21 +295,55 @@ def test_image_file_post_permissions(client, is_active, expected_response):
...
+
+    response = get_view_for_user(
+        viewname="api:upload-session-process-images",
+        reverse_kwargs={"pk": us.pk},
+        user=user,
+        client=client,
+        method=client.patch,
+        content_type="application/json",
+    )
+    assert response.status_code == 400
```
Comment (jmsmkn) [inline app/tests/cases_tests/test_api.py]: This test is just testing if the post save task on `us.save()` works, which we know it does as it is tested elsewhere. What are you intending to test here? Maybe it was restarting the job, in which case 

```
    us.refresh_from_db()
    us.session_state = "created"
    us.save()
```

needs to happen before

```
    ri.refresh_from_db()
    ri.consumed = False
    ri.save()
```

and then 

```
    assert response.status_code == 200
```

as the job will be restarted in this case.

Comment (haimasree) [reply app/tests/cases_tests/test_api.py]: @jmsmkn 
This gives me a 400 response. Why should it be 200? For that, us.session_state must be "stopped". Isn't it?

Comment (jmsmkn) [reply app/tests/cases_tests/test_api.py]: I don't understand what you're trying to do here then, what is the intention of this test?

Comment (haimasree) [reply app/tests/cases_tests/test_api.py]: I guess the last test is superfluous. I will remove it.

**Pull request merged**